src/main/java/com/squareup/kotlinpoet/LambdaTypeName.kt

```diff
@@ -17,7 +17,7 @@ package com.squareup.kotlinpoet
 
 class LambdaTypeName internal constructor(
     val receiver: TypeName? = null,
-    parameters: List<Parameter> = emptyList(),
+    parameters: List<ParameterSpec> = emptyList(),
```

Comment:
Question: should we have checks to verify that all `ParameterSpec`s here have only the name and the type and nothing else?

Reply:
We should check what's supported. I have a feeling annotations might work...

Reply:
Really? Let me see

Reply:
`Unsupported [annotation on parameter in function type]`

Reply:
Modifiers and default values are unsupported as well, hence just name and type